src/execution_plan/execution_plan_reduce_to_apply.c

```diff
@@ -123,13 +84,22 @@ static OpBase *_ReduceFilterToOp(ExecutionPlan *plan,
 
 /* Reduces a filter operation to a SemiApply/ApplyMultiplexer operation, according to the filter tree */
 void ExecutionPlan_ReduceFilterToApply(ExecutionPlan *plan, OpFilter *filter) {
+	// Collect all variables that are bound at this position in the op tree.
+	rax *bound_vars = raxNew();
```

Comment:
I believe `bound_vars` is leaked.
Also I believe `ExecutionPlan_BuildArgumentModifiesArray` should be replaced with a generic `raxKeys` function which returns an array containing a copy of each key in the rax.

Reply:
:+1: Good catch, I don't know how I missed that leak.